core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -211,7 +210,8 @@ public NodeEnvironment(Settings settings, Environment environment) throws IOExce
                             nodePaths[dirIndex] = new NodePath(dir);
                             nodeLockId = possibleLockId;
                         } catch (LockObtainFailedException ex) {
-                            startupTraceLogger.trace("failed to obtain node lock on {}", dir.toAbsolutePath());
+                            startupTraceLogger.trace(
+                                    new ParameterizedMessage("failed to obtain node lock on {}", dir.toAbsolutePath()), ex);
```

Comment:
I think trace is best here, this can arise already in "normal" operations if multiple nodes are started sharing the same data path. Additionally, trace logging is consistent with the remainder of the logging in this class.